op-interop-mon/monitor/updater_test.go

```diff
@@ -0,0 +1,314 @@
+		executingInfo  eth.BlockID
+		initExpiry     eth.NumberAndHash
+		execExpiry     eth.NumberAndHash
+		lastEvaluated  time.Time
+		didMetrics     bool
+		shouldExpire   bool
+	}{
+		{
+			name: "job should expire - both blocks finalized and metrics counted",
+			initiatingInfo: &supervisortypes.Identifier{
+				ChainID:     eth.ChainIDFromUInt64(1),
+				BlockNumber: 100,
+			},
+			executingInfo: eth.BlockID{
+				Number: 200,
+				Hash:   common.HexToHash("0x123"),
+			},
+			initExpiry:    mockNumberAndHash{number: 150, hash: common.HexToHash("0x456")}, // initiating block is finalized
+			execExpiry:    mockNumberAndHash{number: 250, hash: common.HexToHash("0x789")}, // executing block is finalized
+			lastEvaluated: time.Now().Add(-time.Hour),
+			didMetrics:    true,
+			shouldExpire:  true,
+		},
+		{
+			name: "job should not expire - initiating block not finalized",
+			initiatingInfo: &supervisortypes.Identifier{
+				ChainID:     eth.ChainIDFromUInt64(1),
+				BlockNumber: 100,
+			},
+			executingInfo: eth.BlockID{
+				Number: 200,
+				Hash:   common.HexToHash("0x123"),
+			},
+			initExpiry:    mockNumberAndHash{number: 50, hash: common.HexToHash("0x456")},  // initiating block not finalized
+			execExpiry:    mockNumberAndHash{number: 250, hash: common.HexToHash("0x789")}, // executing block is finalized
+			lastEvaluated: time.Now().Add(-time.Hour),
+			didMetrics:    true,
+			shouldExpire:  false,
+		},
+		{
+			name: "job should not expire - executing block not finalized",
+			initiatingInfo: &supervisortypes.Identifier{
+				ChainID:     eth.ChainIDFromUInt64(1),
+				BlockNumber: 100,
+			},
+			executingInfo: eth.BlockID{
+				Number: 200,
+				Hash:   common.HexToHash("0x123"),
+			},
+			initExpiry:    mockNumberAndHash{number: 150, hash: common.HexToHash("0x456")}, // initiating block is finalized
+			execExpiry:    mockNumberAndHash{number: 150, hash: common.HexToHash("0x789")}, // executing block not finalized
+			lastEvaluated: time.Now().Add(-time.Hour),
+			didMetrics:    true,
+			shouldExpire:  false,
+		},
+		{
+			name: "job should not expire - never evaluated",
+			initiatingInfo: &supervisortypes.Identifier{
+				ChainID:     eth.ChainIDFromUInt64(1),
+				BlockNumber: 100,
+			},
+			executingInfo: eth.BlockID{
+				Number: 200,
+				Hash:   common.HexToHash("0x123"),
+			},
+			initExpiry:    mockNumberAndHash{number: 150, hash: common.HexToHash("0x456")},
+			execExpiry:    mockNumberAndHash{number: 250, hash: common.HexToHash("0x789")},
+			lastEvaluated: time.Time{}, // never evaluated
+			didMetrics:    true,
+			shouldExpire:  false,
+		},
+		{
+			name: "job should not expire - metrics not counted",
+			initiatingInfo: &supervisortypes.Identifier{
+				ChainID:     eth.ChainIDFromUInt64(1),
+				BlockNumber: 100,
+			},
+			executingInfo: eth.BlockID{
+				Number: 200,
+				Hash:   common.HexToHash("0x123"),
+			},
+			initExpiry:    mockNumberAndHash{number: 150, hash: common.HexToHash("0x456")},
```

Comment:
Nit: are these hashes actually used for anything during this test? What would happen if the `Hash()` method just returned some fixed hash? Might clean up the test file a bit.